src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAwsUtilityImpl.java

```diff
@@ -0,0 +1,287 @@
+
+import java.io.*;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Optional;
+import java.util.zip.GZIPInputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import ca.corefacility.bioinformatics.irida.exceptions.StorageException;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+import ca.corefacility.bioinformatics.irida.util.FileUtils;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectInputStream;
+
+/**
+ * Component implementation of file utitlities for aws storage
+ */
+@Component
+public class IridaFileStorageAwsUtilityImpl implements IridaFileStorageUtility{
+	private static final Logger logger = LoggerFactory.getLogger(IridaFileStorageAwsUtilityImpl.class);
+
+	private String bucketName;
+	private BasicAWSCredentials awsCreds;
+	private AmazonS3 s3;
+
+	@Autowired
+	public IridaFileStorageAwsUtilityImpl(String bucketName, String bucketRegion, String accessKey, String secretKey){
+		this.awsCreds = new BasicAWSCredentials(accessKey, secretKey);
+		this.s3 = AmazonS3ClientBuilder.standard().withRegion(Regions.fromName(bucketRegion))
+				.withCredentials(new AWSStaticCredentialsProvider(awsCreds)).build();
+		this.bucketName = bucketName;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public IridaTemporaryFile getTemporaryFile(Path file) {
+		try {
+			logger.trace("Getting file from aws s3 [" + file.toString() + "]");
+			Path tempDirectory = Files.createTempDirectory("aws-tmp-");
+			Path tempFile = tempDirectory.resolve(file.getFileName().toString());
+
+			try(S3Object s3Object = s3.getObject(bucketName, getAwsFileAbsolutePath(file))) {
+				S3ObjectInputStream s3ObjectInputStream = s3Object.getObjectContent();
+				org.apache.commons.io.FileUtils.copyInputStreamToFile(s3ObjectInputStream, tempFile.toFile());
+				s3ObjectInputStream.close();
+			} catch (AmazonServiceException e) {
+				logger.error(e.getErrorMessage());
+				throw new StorageException(e.getMessage());
+			} catch (Exception e) {
+				logger.error("Unable to get object from aws S3: " + e);
+				throw new StorageException(e.getMessage());
+			}
+
+			return new IridaTemporaryFile(tempFile, tempDirectory);
+		} catch (FileNotFoundException e) {
+			logger.error(e.getMessage());
+			throw new StorageException(e.getMessage());
```

Comment:
In all these `throw new StorageException` lines can you add the throwable cause to the exception?  That way the  `caused by` lines in the stack trace will actually show us where the exception came from.

So like `throw new StorageException("some message", e)`

Reply:
Updated in 2ff8d49c203b54e1df18e5388736b81378b1323f